core/modules/startup/keyboard.js

```diff
@@ -0,0 +1,97 @@
+/*\
+title: $:/core/modules/startup/keyboard.js
+type: application/javascript
+module-type: startup
+
+Keyboard shortcut handling
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+// Export name and synchronous status
+exports.name = "global-keyboard";
+exports.platforms = ["browser"];
+exports.after = ["startup"];
+exports.synchronous = true;
+
+exports.startup = function() {
+	if($tw.browser) {
+		// Get existing shortcut-tiddlers
+		var shortcutTiddlers = $tw.wiki.filterTiddlers("[all[shadows+tiddlers]tag[$:/tags/KeyboardShortcut]!has[draft.of]]"),
+			shortcutKeysList = [],
+			shortcutActionList = [],
+			shortcutParsedList = [];
+
+		function updateShortcutLists(tiddlerList) {
+			shortcutTiddlers = tiddlerList;
+			for(var i=0; i < tiddlerList.length; i++) {
+				var title = tiddlerList[i],
+					tiddlerFields = $tw.wiki.getTiddler(title).fields;
+				shortcutKeysList[i] = tiddlerFields["key"] !== undefined ? tiddlerFields["key"] : undefined;
```

Comment:
We would usually only use the `["key"]` notation when the string contains characters that aren't legal in JS names; here you can say `tiddlerFields.key` etc.

Reply:
ah yes, I've seen that - I'll change this